src/crawlee/browsers/_playwright_browser_controller.py

```diff
@@ -152,8 +143,45 @@ async def _create_browser_context(self, proxy_info: ProxyInfo | None = None) ->
             else None
         )
 
-        return await self._browser.new_context(
+        if self._use_fingerprints:
+            self._browser_context = await AsyncNewContext(
+                browser=self._browser, fingerprint_options=(fingerprint_options or {}), proxy=proxy
+            )
+            return
+
+        if self._header_generator:
+            common_headers = self._header_generator.get_common_headers()
+            sec_ch_ua_headers = self._header_generator.get_sec_ch_ua_headers(browser_type=self.browser_type)
+            user_agent_header = self._header_generator.get_user_agent_header(browser_type=self.browser_type)
+            headers = dict(common_headers | sec_ch_ua_headers | user_agent_header)
+            extra_http_headers = headers
+            user_agent = headers.get('User-Agent')
+        else:
+            extra_http_headers = None
+            user_agent = None
+
+        self._browser_context = await self._browser.new_context(
             user_agent=user_agent,
             extra_http_headers=extra_http_headers,
             proxy=proxy,
         )
+
+    def _get_browser_context(self) -> BrowserContext:
+        if not self._browser_context:
+            raise RuntimeError('Browser context was not set yet.')
+        return self._browser_context
+
+    @override
```

Comment:
I am confused why code below is shown as changed. There was no change to this in any commit in this PR